src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -1072,6 +1118,166 @@ void DatabaseCatalog::updateLoadingDependencies(const StorageID & table_id, Tabl
     old_dependencies = std::move(new_dependencies);
 }
 
+void DatabaseCatalog::cleanupStoreDirectoryTask()
+{
+    fs::path store_path = fs::path(getContext()->getPath()) / "store";
```

Comment:
@tavplubix But this will not cleanup `store` folder on other disks except default (`path` directive from config), right?

Reply:
Yes, I will add cleanup for other disks

Reply:
#39872